### PR TITLE
fix:always return ActiveRecord relation from SubmissionMatch.find_all

### PR DESCRIPTION
**Pull request by bhoggard**
Props to @mikehrom for the SQL query.

### CODE REVIEW

File: app/models/submission_match.rb [hunk 1]
```diff
@@ -13,20 +13,40 @@ def initialize(params)
...
+      SELECT s.id as submission_id,
+             COUNT(DISTINCT ps.id) AS count_partner_submissions,
+             COUNT(DISTINCT ps.accepted_offer_id) AS count_accepted_offers
+      FROM submissions s
+       LEFT JOIN partner_submissions ps on ps.submission_id = s.id
+      GROUP BY 1
+      )
+      SELECT submission_id
+      from wrap
+      where count_partner_submissions = 0 OR count_accepted_offers = 0
```
Comment (pepopowitz) [inline app/models/submission_match.rb:42]:
```suggestion
    WITH wrap AS (
      SELECT s.id as submission_id,
             COUNT(DISTINCT ps.id) AS count_partner_submissions,
             COUNT(DISTINCT ps.accepted_offer_id) AS count_accepted_offers
      FROM submissions s
       LEFT JOIN partner_submissions ps on ps.submission_id = s.id
      GROUP BY 1
      )
      SELECT submission_id
      FROM wrap
      WHERE count_partner_submissions = 0 OR count_accepted_offers = 0
```

Non-blocking & not a big deal. I'd imagine we don't have any rubocopping on format of SQL queries, so I think we're on our own to manage these kinds of consistencies.

Comment (pepopowitz) [inline app/models/submission_match.rb:42]:
```suggestion
    WITH submissions_with_counts AS (
      SELECT s.id as submission_id,
             COUNT(DISTINCT ps.id) AS count_partner_submissions,
             COUNT(DISTINCT ps.accepted_offer_id) AS count_accepted_offers
      FROM submissions s
       LEFT JOIN partner_submissions ps on ps.submission_id = s.id
      GROUP BY 1
      )
      SELECT submission_id
      from submissions_with_counts
      where count_partner_submissions = 0 OR count_accepted_offers = 0
```

Also non-blocking & not a big deal, but I personally find it helpful for understanding to give contextual names to common table expressions.

**Review by pepopowitz — COMMENTED** [review]
The changes are clear & look good to me, but I don't really understand why they're needed. I would have expected the old code to also be returning an ActiveRecord relation. Can you give more context or link me to some ActiveRecord docs that would make it more clear to me why the old version wasn't working?

File: app/models/submission_match.rb (hunk 1 above)
Comment (mikehrom) [reply app/models/submission_match.rb:42]: +1 to the contextual name 👍

**Comment by bhoggard** [comment]
The old code, by dropping out of `ActiveRecord` code and using methods from `Enumerable`, converted the relation to a regular array.

**Comment by bhoggard** [comment]
#mergeongreen

**Comment by bhoggard** [comment]
Merge on green didn't work. I'm going to merge this and test on staging.

**Pull request merged**